src/database/queries/signup.js

```diff
@@ -0,0 +1,22 @@
+const db_connection = require('../db_connection');
+
+const bcrypt = require('bcrypt');
+
+const register=(userdata,callback) => {
+  const salt=bcrypt.genSaltSync(10);
+  const passwordHash = bcrypt.hashSync(userdata.user_password,salt);
```

Comment:
we should try to use Async so we don't block code. Hashing takes a lot of time, a salt of 10 will take a while to hash the password.

Reply:
By the way, I think we should refactor and modularize our code instead of the additional stretch goals functionality if we have time.